### DISCUSSION TITLE
[Question]: Why is frigate designed to track the whole canvas

### DISCUSSION THREAD

**Discussion by Jebarson** [blakeblackshear/frigate#13108] [Ask A Question] [answered]
### What is your question:

I am curious to understand why Frigate is designed to detect and track whole canvas (visible camera) though I have defined masks and zones. I am piggy backing the detail that I read here https://docs.frigate.video/configuration/masks#further-clarification

Wouldn't drawing a blank opaque mask or only sending the "selected" zone to the detection algorithm be a better way?

Note: I am seriously trying to understand the design and the reason behind it and not questioning it.

**Comment by hawkeye217** [comment]
Some of your question might be answered with a real-world scenario. We just wrote up a new tip yesterday here that might clarify more: https://github.com/blakeblackshear/frigate/discussions/13040

**Reply by Jebarson** [reply]
Thanks for the pointer. While the article explains the difference between mask and zone and why they are not the same, I don't believe it answers the question I have.

For the sake of simplicity, let's take the sample picture that is posted in the article. If I am only interested in the sidewalk, then wouldn't cropping the images pertaining to those boundaries and looking for change be cheaper instead of looking at the whole image?

![image](https://github.com/user-attachments/assets/df91dfba-22b1-47d1-aa15-c464a630862c)

**Reply by hawkeye217** [reply]
Not necessarily. What if a person quickly entered the frame from the right and passed through the zone and into a masked area to the left of the zone? Frigate keeps a score history of detected objects to help determine if an object is a false positive or not. By moving into a masked area (which is effectively the same thing as cropping the image to look at the zone area only), the person may never be detected as a true positive. When they move through the zone and into the center of the frame, Frigate continues to track them, keeping a score history and thus counting them as a true positive and marking them as having passed through the sidewalk zone.

**Reply by Jebarson** [reply]
Lets say we dont have a mask in the camera. I agree with the article that mask and zone are not the same. However, I do feel that if a "required zone" is all that I care about, then I believe that tracking objects outside the area is not necessary.

Or am I missing something obvious?

**Reply by hawkeye217** [reply]
Tracking objects outside the area is necessary for the reasons I just discussed above. 

Say a car moves very quickly across the zone. If Frigate is only tracking cars in the zone, there may not be sufficient frames of the car in the zone to actually mark it as a true positive.

See the docs on how object scores work in Frigate: https://docs.frigate.video/configuration/object_filters/#object-scores

If the car is never marked as a true positive, Frigate will never generate an event for it, and it would appear to you that a car never crossed the zone.

By continuing to track objects outside the zone, Frigate keeps a score history for the object and then saves the footage based on your object retention settings.

**Reply by Jebarson** [reply]
I think I am understanding the reason.

Let me ask a hypothetical question. Let's say instead of monitoring the whole canvas, if frigate were to monitor x+50% and y+50% of the zone area and reduce the number of frames to qualify the detection, do you think it will reduce the strain on the compute but still yield similar accuracy?

**Reply by hawkeye217** [reply, marked as answer]
The entire frame/canvas is rarely sent to the object detector. Frigate uses "regions", a portion of the camera frame that is sent to the object detection pipeline. Regions can be sent due to motion, active objects, or occasionally for stationary objects. These are represented by green boxes in the debug live view. This is of course much more efficient than sending the entire frame.

**Reply by Jebarson** [reply]
Thanks for explaining patiently. Really appreciate it.

**Reply by hawkeye217** [reply]
Happy to help!
